src/pages/login/Login.tsx

```diff
@@ -0,0 +1,118 @@
+/// <reference types="vite-plugin-svgr/client" />
+import { FieldValues, SubmitHandler, useForm } from "react-hook-form";
+import AppIcon from "@/assets/app-icon.svg?react";
+import Modal from "@/components/modal/Modal";
+import React, { useState } from "react";
+import Typography from "@/components/typography/Typography";
+import Button from "@/components/button/Button";
+import { APP_NAME } from "@/common/consts";
+import Input from "@/components/form/input/Input";
+import { useRouter } from "@tanstack/react-router";
+import { useAuth } from "@/hooks/useAuth";
+import { useDataMutation } from "@/hooks/useDataMutation";
+import { User } from "@/models/user/user";
+import { AuthData } from "@/models/auth/authData";
+
+const Login: React.FC = () => {
+  const router = useRouter();
+  const { signIn } = useAuth();
+  const [error, setError] = useState<string | null>(null);
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm();
+
+  const { mutateInstance: mutate } = useDataMutation<User>("/auth", undefined, {
```

Comment:
Writing it once, I believe it should to apply to all the `useDataMutations`
Worth adding another hook, which is simpler - that's called `useLogin` and encapsulate the logic in it- in the future dataMutations would be happening from various places in the app

Reply:
Think about `useAuth`, it's reallty similar